feature_extractor/map_helper.py

```diff
@@ -89,110 +94,163 @@ def update(self, obs, game_state: GameState, entity_helper: EntityHelper):
-    water_arr[-1] = max(0, self.poison_map[row+1, col]) / 20.  # patch after getting trained
-    herb_arr[-1] = max(0, self.poison_map[row, col+1]) / 20.  # patch after getting trained
-    fish_arr[-1] = max(0, self.poison_map[row-1, col]) / 20.  # patch after getting trained
-    obstacle_arr[-1] = max(0, self.poison_map[row, col-1]) / 20.  # patch after getting trained
-    # xcxc
-    return np.zeros(206)
-    return np.concatenate([
-        food_arr, water_arr, herb_arr, fish_arr, obstacle_arr,
-    ])
-
-  def legal_moves(self, obs):
+
+    return np.stack(imgs)
+
+  def nearby_features(self, row: int, col: int, nearby_dist=NEARBY_DIST):
+    # CHECK ME(kywch): my understanding of this function is to provide
+    #   especially important spatial features to the agent
+    #   such as nearby food, water, herb (poultice), fish (ration), obstacles, poison
+    #   As long as the order is fixed, agents will make sense of and use these features
+    near_tile_map = self.tile_map[row-nearby_dist//2:row+(nearby_dist//2+1),
+                                  col-nearby_dist//2:col+(nearby_dist//2+1)]
+    feat_arr = []
+    for i in range(nearby_dist):
+      for j in range(nearby_dist):
+        # (i,j) = (4,4) is the provided (row, col)
+        if abs(i-nearby_dist//2) + abs(j-nearby_dist//2) <= nearby_dist//2:
+          feat_arr.append(near_tile_map[i, j] == material.Forest.index) # food_arr
+          feat_arr.append(near_tile_map[i, j] == material.Water.index) # water_arr
+          feat_arr.append(near_tile_map[i, j] == material.Herb.index) # herb_arr
+          feat_arr.append(near_tile_map[i, j] == material.Fish.index) # fish_arr
+          feat_arr.append(near_tile_map[i, j] in material.Impassible.indices) # obstacle_arr
+
+        # adding poison map, which hadh comment: "patch after getting trained"
+        # CHECK ME: the below was set to <= 0, to make the output len 206
+        #   having wider poison map (like above) may help.
+        #   Changing it will require changing model.py, n_player_feat
+        if abs(i-nearby_dist//2) + abs(j-nearby_dist//2) <= 0: # 1:
+          # CHECK ME: poison_map values can go over 1, unlike the above values
+          feat_arr.append(max(0, self.poison_map[row+i, col+j]) / POISON_CLIP)
+
+    # CHECK ME: the below lines had the comment: "patch after getting trained"
+    #   This looks hacky (and create a blind spot), so I added the poison map feature above
+    #   However, this will a different-sized array.
+    # food_arr[-1] = max(0, self.poison_map[row, col]) / POISON_CLIP
+    # water_arr[-1] = max(0, self.poison_map[row+1, col]) / POISON_CLIP
+    # herb_arr[-1] = max(0, self.poison_map[row, col+1]) / POISON_CLIP
+    # fish_arr[-1] = max(0, self.poison_map[row-1, col]) / POISON_CLIP
+    # obstacle_arr[-1] = max(0, self.poison_map[row, col-1]) / POISON_CLIP
+
+    return np.array(feat_arr)
+
+  def legal_moves(self, obs: Dict[int, Any]):
```

Comment:
I think we should use ActionTargets. However, we can't use ActionTargets for items/inventory since their logic is different. So we might want to change ActionTarget computation to be enabled for just some actions. That way we don't compute targets that we don't use

Reply:
Changed back to using ActionTargets. If so, config.PROVIDE_ACTION_TARGETS must be set to True in main.py, etc.

I'll look into items/inventory and share my thoughts.